Maths/Fibonacci.js

```diff
@@ -1,51 +1,67 @@
-const list = []
-
+// https://en.wikipedia.org/wiki/Generalizations_of_Fibonacci_numbers#Extension_to_negative_integers
 const FibonacciIterative = (nth) => {
-  const sequence = []
+  const sign = nth < 0
+  if (sign) nth = -nth
+  const sequence = [0]
 
   if (nth >= 1) sequence.push(1)
-  if (nth >= 2) sequence.push(1)
+  if (nth >= 2) sequence.push(sign ? -1 : 1)
 
   for (let i = 2; i < nth; i++) {
-    sequence.push(sequence[i - 1] + sequence[i - 2])
+    sequence.push(
+      sign ? sequence[i - 1] - sequence[i] : sequence[i] + sequence[i - 1]
+    )
   }
 
   return sequence
 }
 
+const FibonacciGenerator = function * (negative) {
+  let a = 0
+  let b = 1
+  yield a
+  while (true) {
+    yield b;
+    [a, b] = negative ? [b, a - b] : [b, a + b]
+  }
+}
+
+const list = []
 const FibonacciRecursive = (number) => {
+  const sgn = number < 0
```

Comment:
For consistency, please rename the above variable to `sign` as well. Considering that these are booleans rather than signums `-1` or `1`, I'd use a name like `isNegative`, `neg`, or `negative` here.

Reply:
I'll use `isNeg`, seems like a good compromise between accurate and short